src/main/webapp/app/entities/programming-exercise/programming-exercise-plans-and-repositories-preview.component.ts

```diff
@@ -0,0 +1,68 @@
+                            <li>
+                                <code class="text-white badge badge-primary label-preview"
+                                    >{{ programmingExercise.course?.shortName.toLowerCase() }}{{ programmingExercise.shortName.toLowerCase() }}-exercise</code
+                                >
+                                <jhi-help-icon placement="right" text="artemisApp.programmingExercise.preview.templateRepoTooltip"></jhi-help-icon>
+                            </li>
+                            <li>
+                                <code class="text-white badge badge-primary label-preview"
+                                    >{{ programmingExercise.course?.shortName.toLowerCase() }}{{ programmingExercise.shortName.toLowerCase() }}-solution</code
+                                >
+                                <jhi-help-icon placement="right" text="artemisApp.programmingExercise.preview.solutionRepoTooltip"></jhi-help-icon>
+                            </li>
+                            <li>
+                                <code class="text-white badge badge-primary label-preview"
+                                    >{{ programmingExercise.course?.shortName.toLowerCase() }}{{ programmingExercise.shortName.toLowerCase() }}-tests</code
+                                >
+                                <jhi-help-icon placement="right" text="artemisApp.programmingExercise.preview.testRepoTooltip"></jhi-help-icon>
+                            </li>
+                        </ul>
+                    </div>
+                </div>
+                <div class="col-auto rounded bg-light pt-1 pb-1">
+                    <div>
+                        <span class="font-weight-bold" jhiTranslate="artemisApp.programmingExercise.preview.buildPlans">Build Plans: </span>
+                    </div>
+                    <hr class="pt-0 pb-0 mt-1 mb-1" />
+                    <ul>
+                        <li>
+                            <code class="text-white badge badge-primary label-preview"
+                                >{{ programmingExercise.course?.shortName }}{{ programmingExercise.shortName.toUpperCase() }}-BASE</code
+                            >
+                            <jhi-help-icon placement="right" text="artemisApp.programmingExercise.preview.templateBuildPlanTooltip"></jhi-help-icon>
+                        </li>
+                        <li>
+                            <code class="text-white badge badge-primary label-preview"
+                                >{{ programmingExercise.course?.shortName }}{{ programmingExercise.shortName.toUpperCase() }}-SOLUTION</code
+                            >
+                            <jhi-help-icon placement="right" text="artemisApp.programmingExercise.preview.solutionBuildPlanTooltip"></jhi-help-icon>
+                        </li>
+                    </ul>
+                </div>
+            </div>
+        </ng-container>
```

Comment:
Should we move this to its own file, as this is quite a lot of HTML?

Reply:
True, I moved it to its own file